src/names/preludes.md

```diff
@@ -0,0 +1,144 @@
+> Beginning in the 2018 edition, [use declarations] can reference crates in
+> the extern prelude, so it is considered unidiomatic to use `extern crate`.
+
+> **Note**: Additional crates that ship with `rustc`, such as [`alloc`], and
+> [`test`], are not automatically included with the `--extern` flag when using
+> Cargo. They must be brought into scope with an `extern crate` declaration,
+> even in the 2018 edition.
+>
+> ```rust
+> extern crate alloc;
+> use alloc::rc::Rc;
+> ```
+>
+> Cargo does bring in `proc_macro` to the extern prelude for proc-macro crates
+> only.
+
+<!--
+See https://github.com/rust-lang/rust/issues/57288 for more about the
+alloc/test limitation.
+-->
+
+### The `no_std` attribute
+
+The *`no_std` [attribute]* may be applied at the crate level to prevent the
+[`std`] crate from being automatically added into scope. It does three things:
+
+* Prevents `std` from being added to the [extern prelude](#extern-prelude).
+* Uses [`core::prelude::v1`] in the [standard library prelude] instead of
+  [`std::prelude::v1`].
+* Injects the [`core`] crate into the crate root instead of `std`.
+
+> **Note**: Using the core prelude over the standard prelude is useful when
+> either the crate is targeting a platform that does not support the standard
+> library or is purposefully not using the capabilities of the standard
+> library. Those capabilities are mainly dynamic memory allocation (e.g. `Box`
+> and `Vec`) and file and network capabilities (e.g. `std::fs` and `std::io`).
+
+<div class="warning">
+
+Warning: Using `no_std` does not prevent the standard library from being
+linked in. It is still valid to put `extern crate std;` into the crate and
+dependencies can also link it in.
+
+</div>
+
+## Language prelude
+
+The language prelude includes names of types and attributes that are built-in
+to the language. The language prelude is always in scope. It includes the following:
+
+* [Type namespace]
+    * [Boolean type] — `bool`
+    * [Textual types] — `char` and `str`
+    * [Integer types] — `i8`, `i16`, `i32`, `i64`, `i128`, `u8`, `u16`, `u32`, `u64`, `u128`
+    * [Machine-dependent integer types] — `usize` and `isize`
+    * [floating-point types] — `f32` and `f64`
+* [Macro namespace]
+    * [Built-in attributes]
+
+## `macro_use` prelude
+
+The `macro_use` prelude includes macros from external crates that were
+imported by the [`macro_use` attribute] applied to an [`extern crate`].
+
+## Tool prelude
+
+The tool prelude includes tool names for external tools in the [type
+namespace]. See the [tool attributes] section for more details.
+
+## The `no_implicit_prelude` attribute
+
+The *`no_implicit_prelude` [attribute]* may be applied at the crate level or
+on a module to indicate that it should not automatically bring the [standard
+library prelude], [extern prelude], or [tool prelude] into scope for that
+module or any of its descendants.
+
+This attribute does not affect the [language prelude] or [`macro_use` prelude].
```

Comment:
It affects `macro_use` prelude on >=2018 edition.
The plan was to exclude `macro_use` prelude on both editions (https://github.com/rust-lang/rust/pull/55630), but I never got to it.

Reply:
Oops, sorry I missed that. For various reasons, I do all my testing in 2015, and I didn't think to try 2018. I even saw that code, and somehow forgot to double-check it. I thought it was weird that it didn't affect `macro_use`.